packages/block-renderer/src/hooks/use-pattern-inline-css.tsx

```diff
@@ -0,0 +1,29 @@
+import { shuffle } from '@automattic/js-utils';
+import { useMemo } from 'react';
+import { MAX_BLOG_POSTS } from '../constants';
+
+// Using this object to memoize the pattern as a workaround
+const memoPatternOrder: { [ key: string ]: string } = {};
+
+const usePatternInlineCss = ( patternId: string ) => {
+	// FIXME: Why this useMemo is not working?
```

Comment:
What do you mean by not working? 🤔

Reply:
I expect that every time the hook is called with the same `patternId`, the memo will return the same `inlineCss`.

As a result, I expect the order of posts in the pattern preview in the canvas and thumbnail to be the same.

I tested it by commenting out this return:

```
if ( inlineCss ) {
  //return inlineCss;
}
```

And the order of posts is not the same in both previews:

<img width="940" alt="Screenshot 2566-10-03 at 10 46 15" src="https://github.com/Automattic/wp-calypso/assets/1881481/8f0e8c72-d26c-4e41-a8cf-f4485ece7929">

Reply:
It's expected as they are different components 😅

Reply:
Thanks for confirming it. I was guessing that but wasn't sure.

Do you think the workaround that uses `memoPatternOrder` is okay?

What would be the recommended way to share this state between components? A context?

Reply:
I think it's fine to use `memoPatternOrder` here 🙂

Reply:
Yes, I also think it is fine.